DTDL/v2/samples/Diagnostics.json

```
@@ -0,0 +1,26 @@
+{
+    "@context": "dtmi:dtdl:context;2",
+    "@id": "dtmi:com:example:Diagnostics;1",
+    "@type": "Interface",
+    "displayName": "Diagnostics",
+    "description": "Provides functionality to report memory usage and reboot the device.",
+    "contents": [
+      {
+        "@type": "Telemetry",
+        "description": "Current workingset of the device application",
+        "displayName": "Workingset",
+        "name": "workingset",
+        "schema": "long"
```

>long [](start = 19, length = 4)

Does this need to be a long? As you know, when using some languages/runtimes (i.e. JavaScript), long values silently lose precision.

---

Long a 'delicate' story.... integer is large enough if we talk in KB

---

+1.  DeviceInfo used KB specifically so there'd be headroom.  And I think we need a `double` here though.  Even though a fractional KB is strange, my rough math is that a signed-integer of 2^31 KB would correspond to 2 TB.  Which feels very unlikely for embedded (we'll run with way less:)) but if someone followed this pattern for storage available 2TB is legit.